src/backend/bpf.rs

```diff
@@ -0,0 +1,213 @@
+//! Define constants, helpers and types used for BPF codegen.
+
+use super::TargetArch;
+
+/// Program made up of a sequence of BPF instructions.
+pub type BpfProgram = Vec<sock_filter>;
+
+/// Reference to program made up of a sequence of BPF instructions.
+pub type BpfProgramRef<'a> = &'a [sock_filter];
+
+// The maximum number of a syscall argument.
+// A syscall can have at most 6 arguments.
+// Arguments are numbered from 0 to 5.
+pub const ARG_NUMBER_MAX: u8 = 5;
+
+// The maximum number of BPF statements that a condition will be translated into.
+// This is not a linux requirement but is based on the current BPF compilation logic.
+// This is used in the backend code for preallocating vectors and for detecting unjumpable offsets.
+pub const CONDITION_MAX_LEN: u8 = 6;
+
+// The maximum seccomp-BPF program length allowed by the linux kernel.
+pub const BPF_MAX_LEN: usize = 4096;
+
+// `struct seccomp_data` offsets and sizes of fields in bytes:
+//
+// ```c
+// struct seccomp_data {
+//     int nr;
+//     __u32 arch;
+//     __u64 instruction_pointer;
+//     __u64 args[6];
+// };
+// ```
+pub const SECCOMP_DATA_NR_OFFSET: u8 = 0;
+const SECCOMP_DATA_ARCH_OFFSET: u8 = 4;
+pub const SECCOMP_DATA_ARGS_OFFSET: u8 = 16;
+pub const SECCOMP_DATA_ARG_SIZE: u8 = 8;
+
+// Builds a `jump` BPF instruction.
+//
+// # Arguments
+//
+// * `code` - The operation code.
+// * `jt` - The jump offset in case the operation returns `true`.
+// * `jf` - The jump offset in case the operation returns `false`.
+// * `k` - The operand.
+#[allow(non_snake_case)]
```

Comment:
What is the reason for naming these functions with upper case? They are not const.

Reply:
I don't think that keeping the upper case name helps in this case. The reason it is written in uppercase in the kernel is that it is a macro, and the C convention says that macros are all caps. Since this is Rust code, we should follow the Rust convention.

When reading the code you are misguided because what seems to be a constant, is actually a function.

Reply:
I think there is no right or wrong in this case. Really I'm fine either way

Reply:
I've lowercased the names